### PR TITLE
test(ice): stress-test non-routable peer filter under flood (#2386)

### PR DESCRIPTION
**Pull request by YPandas**
*Issue #, if available:*

- #2386 (TURN `403 Forbidden IP` stalls ICE gathering). Adds stress/scale coverage for the non-routable peer filter delivered in #2388.

*What was changed?*

- Added two injection/stress unit tests to `tst/TurnConnectionFunctionalityTest.cpp` for the non-routable TURN peer filter:
  - `turnConnectionFiltersDistinctNonRoutableFloodFast`
  - `turnConnectionFloodDoesNotDelayRoutablePeers`
- Added shared `allocStressTestTurnConnection` / `freeStressTestTurnConnection` test helpers.
- Test-only change; no production code modified.

*Why was it changed?*

- The filter's whole point is that a large number of local/non-routable peer candidates can be rejected up front so they never trigger a `CreatePermission` the TURN server would answer with `403 Forbidden IP` (which otherwise stalls the state machine until the ~5s-per-state timeout). Prior tests confirmed the single-peer behavior but not that the filter holds up under scale.
- These lock in two properties a regression could silently break: (1) a huge flood of distinct non-routable peers is absorbed cheaply and never enters the peer list, and (2) a flood does not crowd out or delay genuinely routable peers — the state machine still advances immediately once the real peers succeed.
- Per-range coverage of the classifier itself already lives in `NetworkApiTest.IsNonRoutableAddr*`, so these tests deliberately focus on scale and flood behavior rather than duplicating range checks.

*How was it changed?*

- Both tests build a minimal heap `TurnConnection` (live mutex, `stateTimeoutTime` pushed far into the future so results reflect peer-state counting rather than a timeout fallthrough) and drive `turnConnectionAddPeer` directly — no network or credentials, so they are not gated on `mAccessKeyIdSet` and always run.
- `turnConnectionFiltersDistinctNonRoutableFloodFast` injects 500,000 **distinct** non-routable IPv4 addresses (packed across `10.0.0.0/8`) and asserts every call is a successful no-op, `turnPeerCount == 0`, `nonRoutablePeersFilteredCount == 500000`, and the flood completes well under a single per-state timeout.
- `turnConnectionFloodDoesNotDelayRoutablePeers` interleaves 100,000 non-routable peers with 3 genuinely routable ones, then asserts only the routable peers are added (`turnPeerCount == 3`, the rest filtered) and that `fromCreatePermissionTurnState` advances to `TURN_STATE_BIND_CHANNEL` once they succeed.
- The per-skip `INFO` log is silenced during the flood (`SET_LOGGER_LOG_LEVEL(LOG_LEVEL_WARN)`, restored to `mLogLevel`) so timing reflects filter cost, not log I/O; teardown frees each added peer's transaction-id store to keep the CI sanitizer jobs clean.

*What testing was done for the changes?*

- Built `webrtc_client_test` locally (`-DBUILD_TEST=ON`) and ran both new cases:
  `./tst/webrtc_client_test --gtest_filter='TurnConnectionFunctionalityTest.turnConnectionFiltersDistinctNonRoutableFloodFast:TurnConnectionFunctionalityTest.turnConnectionFloodDoesNotDelayRoutablePeers'` — 2/2 passed.
- Confirmed the functional assertions are the real regression guards (if the filter were removed, adds would fill the 32-peer cap then return errors, `turnPeerCount` would be non-zero, and `nonRoutablePeersFilteredCount` would be 0 — all caught independently of timing).
- Tests are pure logic (no network/credentials), so they run identically in CI across the OS/TLS/compiler build matrix, including the sanitizer jobs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tst/TurnConnectionFunctionalityTest.cpp [hunk 1]
```diff
@@ -1177,6 +1177,159 @@ TEST_F(TurnConnectionFunctionalityTest, turnConnectionCallMultipleTurnSendDataIn
...
+        // timed region reflects only the filter cost.
+        allFiltered = (BOOL) (allFiltered && (turnConnectionAddPeer(pTurnConnection, &peer) == STATUS_SUCCESS));
+    }
+    UINT64 floodElapsed = GETTIME() - startTime;
+
+    SET_LOGGER_LOG_LEVEL(mLogLevel);
+
+    EXPECT_TRUE(allFiltered);
+    // None of the flood entered the peer list; every one was counted as filtered.
+    EXPECT_EQ((UINT32) 0, pTurnConnection->turnPeerCount);
```
Comment (sirknightj) [inline tst/TurnConnectionFunctionalityTest.cpp:1260]: Is there a difference between what this test is testing vs. the tests added in #2388? Both check that the filter filters the invalid ones.

Is there a need to test 500k? Seems like 1 (since we're checking non-zero) would suffice. In case of a slow environment, it may not finish in time.

The delays in the issue comes from the network retries/delay, but there are no network calls in these tests, so the 3s timeout is mostly testing the CPU speed.

Comment (YPandas) [reply tst/TurnConnectionFunctionalityTest.cpp:1260]: I originally wanted to sanity-check that throwing a very large number of non-routable candidates at `turnConnectionAddPeer` gets absorbed cleanly (all filtered, none added) rather than accumulating. But you're right that peer #500k exercises the same code as peer #1 and the big count probably only adds CI flakiness risk on slow/sanitizer runners without covering anything new.

I'll close this PR since the same tests has already been done.
